unlock-app/src/services/currencyLookupService.js

```diff
@@ -0,0 +1,30 @@
+import axios from 'axios'
+
+export default class CurrencyLookupService {
+  constructor(uri) {
+    this.host = new URL(uri).host
+  }
+
+  async lookupPrice(base, currency) {
+    if (RegExp('api.coinbase.com', 'ig').test(this.host)) {
+      return this._handleCoinbaseFetch(base, currency)
+    } else {
+      throw 'Unknown Currency Conversion Provider'
+    }
+  }
+
+  _constructCoinbaseLookupURI(base, currency) {
+    return `https://api.coinbase.com/v2/prices/${base.toUpperCase()}-${currency.toUpperCase()}/buy`
+  }
+
+  async _handleCoinbaseFetch(base, currency) {
+    let result = await axios.get(
+      this._constructCoinbaseLookupURI(base, currency)
+    )
+
+    return Promise.resolve({
```

Comment:
I think you can just return the value and it uses it as the return in .then?

Reply:
I think I can have it implicitly return `result.data.data`, the major reason for doing it was to shape the data that the middleware should expect from the service basically a object with only these two properties: currency & amount

Reply:
No I mean you don't need the Promise.resolve() around it, I like that you're explicit, sorry I was unclear!

Reply:
I agree we should not return `result.data.data` which is specific to Axios.
Now indeed like @cellog the Promise.resolve is not required?